filters/include/pcl/filters/experimental/voxel_grid.h

```diff
@@ -26,22 +26,22 @@ struct Voxel {
     Centroid() {}
     ~Centroid() {}
 
-    CentroidPoint<PointT> all_fields;
+    CentroidPoint<PointT>* all_fields;
```

Comment:
unique_ptr instead?

Reply:
segfault during move assign again, it throws error while resetting itself

Reply:
Definitely needs to be solved properly. Perhaps after GSoC. Please make a big note in the file. Current implementation is not copy-safe